provider/source/src/time_zones/mod.rs

```diff
@@ -150,11 +150,16 @@ impl SourceDataProvider {
                     .meta_zone_ids
                     .0;
 
-                let mut meta_zone_ids = BTreeMap::new();
-                for (meta_zone_id, meta_zone_id_data) in meta_zone_ids_resource.iter() {
-                    meta_zone_ids.insert(meta_zone_id_data.long_id.to_string(), *meta_zone_id);
-                }
-                Ok(meta_zone_ids)
+                Ok(meta_zone_ids_resource
+                    .iter()
+                    .enumerate()
+                    .map(|(idx, (_short_id, alias))| {
+                        (
+                            alias.long_id.to_string(),
+                            MetazoneId::new(idx as u8 + 1).unwrap(),
```

Comment:
Issue: is this where you convert from string to number? If we do this type of change, we should define the numbers as a fixed mapping, not dependent on cldr version. If you want to depend on version, add a hash of the list of strings to the data.